plugin-taskbar/lxqttaskbar.cpp

```diff
@@ -304,11 +305,33 @@ void LXQtTaskBar::addWindow(WId window)
         connect(group, &LXQtTaskButton::dragging, this, [this] (QObject * dragSource, QPoint const & pos) {
             buttonMove(qobject_cast<LXQtTaskGroup *>(sender()), qobject_cast<LXQtTaskGroup *>(dragSource), pos);
         });
-
         mLayout->addWidget(group);
         group->setToolButtonsStyle(mButtonStyle);
-    }
+        const QString window_class = QString::fromUtf8(KWindowInfo(window, NET::Properties(), NET::WM2WindowClass).windowClassClass());
+        int src_index = mLayout->count() - 1;
+        int dst_index = src_index;
+        if (mUngroupedNextToExisting)
+        {
```

Comment:
Please put the variables `window_class`, `src_index` and `dst_index` inside the conditional block, above the `for` loop.